src/vcpkg/base/messages.cpp

```diff
@@ -95,17 +96,160 @@ namespace vcpkg
+    static const char *loaded_localization_file_begin = 0, *loaded_localization_file_end = 0;
+
+    namespace
+    {
+        enum class message_index
+        {
+#define DECLARE_MESSAGE(NAME, ARGS, COMMENT, ...) NAME,
+#include <vcpkg/base/message-data.inc.h>
+#undef DECLARE_MESSAGE
+        };
+    }
+
+    namespace msg
+    {
+        namespace
+        {
+            std::string get_localization_comment(::size_t index)
+            {
+                if (index >= detail::number_of_messages) Checks::unreachable(VCPKG_LINE_INFO);
+                std::string msg = message_data[index].comment.to_string();
+                for (auto&& ex : message_data[index].arg_examples)
+                {
+                    if (ex == nullptr || ex->empty()) continue;
+                    if (!msg.empty()) msg.push_back(' ');
+                    msg.append("An example of ");
+                    msg.append(ex->data(), ex->size());
+                }
+                return msg;
+            }
+        }
+
+        std::vector<RawMessage> get_sorted_raw_messages()
+        {
+            struct MessageSorter
+            {
+                bool operator()(const RawMessage& lhs, const RawMessage& rhs) const { return lhs.name < rhs.name; }
+            };
+
+            std::vector<RawMessage> messages(msg::detail::number_of_messages);
+            for (::size_t index = 0; index < msg::detail::number_of_messages; ++index)
+            {
+                auto& msg = messages[index];
+                msg.name = message_data[index].name.to_string();
+                msg.value = message_data[index].builtin_message.to_string();
+                msg.comment = get_localization_comment(index);
+            }
+            std::sort(messages.begin(), messages.end(), MessageSorter{});
+            return messages;
+        }
 
+        LocalizedString detail::format_message_by_index(size_t index, fmt::format_args args)
+        {
+            if (index >= detail::number_of_messages) Checks::unreachable(VCPKG_LINE_INFO);
+            try
+            {
+                if (loaded_localization_data)
+                {
+                    return LocalizedString::from_raw(fmt::vformat(loaded_localization_data[index], args));
+                }
+            }
+            catch (...)
+            {
+            }
+            const auto default_format_string = message_data[index].builtin_message;
+            try
+            {
+                return LocalizedString::from_raw(
+                    fmt::vformat({default_format_string.data(), default_format_string.size()}, args));
+            }
+            catch (...)
+            {
+            }
+            ::fprintf(stderr,
+                      "INTERNAL ERROR: failed to format default format string for index %zu\nformat string: %.*s\n",
+                      index,
+                      (int)default_format_string.size(),
+                      default_format_string.data());
```

Comment:
```suggestion
            msg::write_unlocalized_text_to_stdout(Color::error, fmt::format("INTERNAL ERROR: failed to format default format string for index {}\nformat string: {}\n",
                      index,
                      default_format_string));
```

I think we want all output going through write_Xxx so that we don't have to worry about interleaving with the CRT